frontend/src/components/MiniCal/MiniCal.tsx

```diff
@@ -103,7 +106,11 @@ const MiniCal = () => {
               <button
                 className={cn(styles.btn2, { [styles.active]: isToday(date) })}
                 onClick={() => {
-                  updateDate(new Date());
+                  const today = new Date();
+                  if (datePickerRef.current) {
+                    datePickerRef.current.setSelected(today);
+                  }
```

Comment:
Optional chaining works with function calls too; also `setPreSelection` seems more appropriate than `setSelected` here (likely less overhead). It's also what's used in [`react-datepicker`'s native implementation of the "Today" button](https://github.com/Hacker0x01/react-datepicker/blob/main/src/calendar.jsx#L731).

```suggestion
                  datePickerRef.current?.setPreSelection?.(today);
```